pkg/metrics/cluster/node.go

```diff
@@ -114,15 +114,13 @@ func (n *node) ApplyConfig(cfg Config) error {
 	if err != nil {
 		return fmt.Errorf("failed to create ring: %w", err)
 	}
-	if err := n.reg.Register(r); err != nil {
-		return fmt.Errorf("failed to register ring metrics: %w", err)
-	}
+
 	if err := services.StartAndAwaitRunning(context.Background(), r); err != nil {
 		return fmt.Errorf("failed to start ring: %w", err)
 	}
 	n.ring = r
 
-	lc, err := ring.NewLifecycler(cfg.Lifecycler, n, "agent", agentKey, false, n.reg)
+	lc, err := ring.NewLifecycler(cfg.Lifecycler, n, "agent", agentKey, false, n.log, prometheus.WrapRegistererWithPrefix("cortex_", n.reg))
```

Comment:
Random Q: I assume this is for backward compatibility but should we change this from `cortex_` to something more agent specific?

Reply:
That's a good question -- I'm not sure. This is just added for backward compatibility as you noted.